core/src/test/scala/unit/kafka/raft/RaftManagerTest.scala

```diff
@@ -221,16 +225,23 @@ class RaftManagerTest {
   @Test
   def testUncaughtExceptionInIoThread(): Unit = {
     val raftClient = mock(classOf[KafkaRaftClient[String]])
-    val ioThread = new RaftIoThread(raftClient, threadNamePrefix = "test-raft")
+    val faultHandler = mock(classOf[FaultHandler])
```

Comment:
easier to just use `MockFaultHandler`

you can then look at `MockFaultHandler.firstException` to at least see if it has the `Unexpected error in raft IO thread` text